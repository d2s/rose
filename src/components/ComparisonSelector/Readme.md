### ComparisonSelector example

Default comparison selector element

```js
<ComparisonSelector />
```
