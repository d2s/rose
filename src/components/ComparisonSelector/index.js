import React, { Component } from 'react';
import PropTypes from 'prop-types';
// import glamorous, {ThemeProvider} from 'glamorous';

// Currency comparison selector
// const ComparisonSelectorStyles = glamorous.div({
//   display: 'inline-block',
//   marginLeft: 'auto',
// });

/**
 * ComparisonSelector component for selecting what things to compare to each other.
 */
class ComparisonSelector extends Component {
  static propTypes = {
    /** Base currency in the comparison. */
    fromCurrency: PropTypes.string,
    /** Other currency to compare the base currency with. */
    toCurrency: PropTypes.string
  }
  static defaultProps = {
    fromCurrency: 'EUR',
    toCurrency: 'USD'
  }

  render() {
    return (
      <div>
        from {this.props.fromCurrency} to [{this.props.toCurrency}]
      </div>
    );
  }
}

export default ComparisonSelector;
