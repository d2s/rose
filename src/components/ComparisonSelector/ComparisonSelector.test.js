import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer'

import ComparisonSelector from './index';

describe('<ComparisonSelector />', () => {
  describe('with ReactDOM', () => {
    it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<ComparisonSelector />, div);
      ReactDOM.unmountComponentAtNode(div);
    });
  });

  describe('with Enzyme', () => {
    it('shallow renders without crashing', () => {
      shallow(<ComparisonSelector />);
    });
  });

  describe('with Snapshot Testing', () => {
    it('renders the same way', () => {
      const component = renderer.create(<ComparisonSelector />)
      const tree = component.toJSON()
      expect(tree).toMatchSnapshot()
    });
  });
});
