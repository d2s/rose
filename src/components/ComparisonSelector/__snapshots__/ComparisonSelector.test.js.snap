// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComparisonSelector /> with Snapshot Testing renders the same way 1`] = `
<div>
  from 
  EUR
   to [
  USD
  ]
</div>
`;
